### ISSUE TITLE
no-unused-variable doesn't properly handle "declare" vars

### ISSUE THREAD

**Issue by derekcicerone** [palantir/tslint#190]
For example, this should not be flagged as unused:
```
declare var assert: chai.Assert;
```

**Comment by filipstachura** [comment]
+1

**Comment by marekrogala** [comment]
+1

**Comment by filipstachura** [comment]
any updates here?

**Comment by ashwinr** [comment]
we'll take a look at it as part of the 1.5 migration, should be straightforward.

**Comment by adidahiya** [comment]
It feels a little weird to me to not use a declared variable; I could see a case for this being an _optional_ part of the `no-unused-variable` rule.

Is this because you're declaring a variable in one source file and using it in another? Or are you running tslint on your typings files? In the latter case, we'd advise only running tslint on source code.

**Comment by filipstachura** [comment]
We declare in one file and then reference it in others. That way when we refactor the name of injected global compilation fails in all files that were not updated.
